modelhub/modelhub/pipelines/extracted_contexts.py

```
@@ -99,17 +100,65 @@ class ExtractedContextsPipeline(BaseDataPipeline):
+            self._taxonomy_column = _get_taxonomy_column_definition(engine)
+            self._base_dtypes = {
+                self._taxonomy_column.name: self._taxonomy_column.dtype,
+                **self.required_context_columns_per_dialect[self._db_dialect],
+            }
+        else:
+            # No taxonomy column, we need to gather the contexts separately.
+            if not is_bigquery(engine):
+                raise Exception('only bigquery supports new data format for now')
+
+            base_dtypes: Dict[str, Any] = {
+                **self.required_context_columns_per_dialect[self._db_dialect],
+                'event_id': bach.SeriesString.dtype,
+                'network_userid': bach.SeriesString.dtype,
+                'se_action': bach.SeriesString.dtype,
+                'se_category': bach.SeriesString.dtype,
+                'true_tstamp': bach.SeriesTimestamp.dtype
+            }
+
+            global_contexts_columns = {}
+
+            for col, dtype in dtypes.items():
+                # match things like contexts_io_objectiv_context_cookie_id_context_1_0_0
+                # "contexts_io_objectiv_" + context_data_name + "_" + ve_rs_ion
+                if not col.startswith('contexts_io_objectiv_'):
+                    continue
+
+                parts = col.split('_')
+                data_name = parts[3:-3]
+                # The version is contained in parts[-3:]
+
+                if data_name[0] == 'context':
+                    context_name = "_".join(data_name[1:-1])
+                    if context_name in global_contexts:
+                        base_dtypes[col] = [{}]
+                        global_contexts_columns[context_name] = col
+
+                elif data_name[:2] == ['location', 'stack']:
+                    base_dtypes[col] = [{'location_stack': bach.SeriesString.dtype}]
+                    global_contexts_columns['location_stack'] = col
+
+            self._base_dtypes = base_dtypes
```

We should probably raise here if some taxonomy column was not found.